calendar_reboot.js

```diff
@@ -689,39 +691,114 @@ Calendar.prototype.removeActiveCalendar = function() {
         
         var self = this;
         //store the state in localStorage
-        storeInLocalStorage(self.calendarState.uniqueId, self.calendarState);
+        storeInLocalStorage(self.state.uniqueId, self.state);
         
-        //put calendar in calendarUniqueIds and store it
-        calendarUniqueId[self.calendarState.uniqueId] = self.calendarState.calendarTitle;
-        storeInLocalStorage(masterKey, calendarUniqueId);
+        //put calendar in allCalendarIdss and store it
+        allCalendarIds[self.state.uniqueId] = self.state.title;
+        storeInLocalStorage(allCalendarIdsKey, allCalendarIds);
         
 };
     
 Calendar.prototype.deleteCalendar = function() {
     //deletes the calendar from localStorage, deletes the calendar from the
-    //calendarUniqueId object
+    //allCalendarIds object
     
     var self = this;
-    //delete the calendar from the calendarUniqueId object
-    delete calendarUniqueId[calendar.calendarState.uniqueId];
+    //delete the calendar from the allCalendarIds object
+    delete allCalendarIds[self.state.uniqueId];
     //save that change in localstorage
-    storeInLocalStorage(masterKey, calendarUniqueId);
+    storeInLocalStorage(allCalendarIdsKey, allCalendarIds);
     
     //remove the calendarstate from localStorage
-    removeFromLocalStorage(calendar.calendarState.uniqueId);
+    removeFromLocalStorage(self.state.uniqueId);
 
 }
     
 
 //dictionary of savedCalendars, calendar title: unique ID
 
-var calendarUniqueId = {
+var allCalendarIds = {
 };
 
+//Make a storage manager
+var LocalCalendarStorage = function(params) {
+    var self = this;
+    var prefix = params['storeId'] || "";
+    
+    //params.storeId
+    
+    var toKey = function(id) {
+        //make a key out of a uniqueId
+        
+        return prefix + "_" + id;
+        
+    };
+    
+    self.getAllCalendarIds = function() {
+        //gets the allCalendarIds object from storage
+        
+        return loadFromLocalStorage(toKey(allCalendarIdsKey));
+    };
+    
+    self.save = function(appObj) {
+        //save an App object (like a calendar object for example) in storage
+        
+        //store the state in localStorage
+        storeInLocalStorage(toKey(appObj.state.uniqueId), appObj.state);
+        
+        //put calendar in allCalendarIdss and store it
+        allCalendarIds[appObj.state.uniqueId] = appObj.state.title;
+        storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
+    };
+    
+    self.remove = function(appObj) {
+        //remove an app object (like a calendar) from storage
+        
+        //delete the calendar from the allCalendarIds object
+        delete allCalendarIds[appObj.state.uniqueId];
+        //save that change in localstorage
+        storeInLocalStorage(toKey(allCalendarIdsKey), allCalendarIds);
+        
+        //remove the object's state from localStorage
+        removeFromLocalStorage(toKey(appObj.state.uniqueId));
+    };
```

Comment:
Then you have to do removeActiveStatus after, which is also a storage manager method.